lsp-haskell.el

```diff
@@ -99,17 +105,17 @@ For example, use the following the start the hie process in a nix-shell:
              :text ,newname))))
 
 (defun lsp-if-to-case ()
-  "Convert an if statement to a case statement"
+  "Convert an if statement to a case statement."
   (interactive)
   (lsp--cur-workspace-check)
   (lsp--send-execute-command
    "hare:iftocase"
    (vector `(:file      ,(concat "file://" buffer-file-name)
-             :start_pos ,(lsp-get-start-position)
-             :end_pos   ,(lsp-get-end-position)))))
+             :start_pos ,(lsp--region-or-line)
+             :end_pos   ,(lsp--region-or-line)))))
```

Comment:
We are trying to have a position for the start of the range, and another one for the end.

The change calls the same function for both positions.  Is this correct?

Reply:
No, you are right, I didn't pay attention.
Do you know the correct "lap" way of doing that? given that `lsp-get-start-posiiton` and `lsp-get-end-position` doesn't exists anymore

Reply:
@ericdallo From poking around older (ca. 2018) commits, here's what they had at the time:

``` elisp
(defun lsp--get-start-position ()
  "Get the start of the region if active, else current point."
  (let ((pos (if (use-region-p)
                 (region-beginning)
               (point))))
    (lsp-point-to-position pos)))

(defun lsp--get-end-position ()
  "Get the end of the region if active, else current point."
  (let ((pos (if (use-region-p)
                 (region-end)
               (point))))
    (lsp-point-to-position pos)))

(defalias 'lsp-get-start-position #'lsp--get-start-position)
(defalias 'lsp-get-end-position #'lsp--get-end-position)
```